src/common/charts/shared.js

```diff
@@ -0,0 +1,99 @@
+            // refer to https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+            d: 'M0 0 L0 3', // draw a 3 until vertical line
+        },
+        width: 4.5, // defines the x bounds of the repeating pattern
+        height: 3, // defines the  y bounds of the repeating pattern
+        backgroundColor: chartTertiaryBackgroundColor, // sets the patterns background color
+        color: chartTertiaryStrokeColor, // sets the patterns stroke color
+        patternTransform: 'rotate(-60)', // rotates the path -60 degrees
+    },
+};
+export const pattern2 = {
+    pattern: {
+        path: {
+            // d is a standard SVG path definition string
+            // refer to https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+            d: 'M0 0 L3 0',
+        },
+        width: 3, // defines the x bounds of the repeating pattern
+        height: 5, // defines the  y bounds of the repeating pattern
+        backgroundColor: chartQuaternaryBackgroundColor, // sets the patterns background color
+        color: chartQuaternaryStrokeColor, // sets the patterns stroke color
+    },
+};
+// configure the color sequences based on how many series are used
+export const colorsSingle = [chartPrimaryColor];
+export const colorsTwo = [chartSecondaryColor, chartPrimaryColor];
+export const colorsThree = [chartSecondaryColor, chartPrimaryColor, pattern1];
+export const colorsMore = [
+    chartSecondaryColor,
+    pattern1,
+    chartPrimaryColor,
+    pattern2,
+    chartQuinaryBackgroundColor,
+];
+// function is used to set up the colors including lineColor(svg stroke) on each of the series objects
+// based on the length of the series array
+export function setSeriesColors(series) {
+    let colors;
+    switch (series.length) {
+        case 1:
+            colors = colorsSingle;
+            series[0].lineColor = series[0].borderColor = chartPrimaryColor;
+            break;
+        case 2:
+            colors = colorsTwo;
+            series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+            series[1].lineColor = series[1].borderColor = chartPrimaryColor;
+            break;
+        case 3:
+            colors = colorsThree;
+            series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+            series[1].lineColor = series[1].borderColor = chartPrimaryColor;
+            series[2].lineColor = series[2].borderColor = chartTertiaryStrokeColor;
+            break;
+        default:
+            colors = colorsMore;
+            series[0].lineColor = series[0].borderColor = chartSecondaryColor;
+            series[1].lineColor = series[1].borderColor = chartTertiaryStrokeColor;
+            series[2].lineColor = series[2].borderColor = chartPrimaryColor;
+            if (series.length > 3) {
+                series[3].lineColor = series[3].borderColor = chartQuaternaryStrokeColor;
+            }
+            if (series.length > 4) {
```

Comment:
This was done to prevent code duplication.  as I need the lines 87-90 to be executed for both and didn't think it was worth trying to break out into another function call